### PR TITLE
feat: onchain endpoint

### PR DESCRIPTION
**Pull request by akhercha**
## Added
- base onchain endpoint:  `/node/v1/onchain/{base}/{quote}`

### CODE REVIEW

File: pragma-node/src/infra/repositories/onchain_repository.rs [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+// TODO(akhercha): Does not give the same result than onchain call
+pub async fn get_last_updated_timestamp(
+    pool: &Pool,
+    network: Network,
+    pair_id: String,
+) -> Result<u64, InfraError> {
+    let raw_sql = format!(
+        r#"
+        SELECT 
+            *
```
Comment (EvolveArt) [inline pragma-node/src/infra/repositories/onchain_repository.rs:139]: shouldn't we just select timestamp field ?

File: infra/pragma-ingestor/config.yml [hunk 2]
```diff
@@ -5,7 +5,7 @@ container_environment:
   - region: "eu-west-3"
   - prefix: "/conf/{{ SERVICE_NAME }}/{{ RUN_ENV }}"
   - keys:
-      - DATABASE_URL
+      - POSTGRES_DATABASE_URL
```
Comment (EvolveArt) [inline infra/pragma-ingestor/config.yml:8]: ingestor service only interacts with Timescale

File: infra/pragma-node/config.yml [hunk 3]
```diff
@@ -5,7 +5,7 @@ container_environment:
   - region: "eu-west-3"
   - prefix: "/conf/{{ SERVICE_NAME }}/{{ RUN_ENV }}"
   - keys:
-      - DATABASE_URL
+      - POSTGRES_DATABASE_URL
```
Comment (EvolveArt) [inline infra/pragma-node/config.yml:8]: we need the two dbs in pragma-node

File: infra/pragma-node/Dockerfile [hunk 4]
```diff
@@ -26,7 +26,7 @@ USER node:pragma
 WORKDIR /home/pragma-node
 
 COPY --from=builder --chown=node:pragma  /home/pragma-node/bin/pragma-node /usr/local/bin/pragma-node
-# COPY infra/pragma-node/config/.env.example .env
+COPY infra/pragma-node/config/.env .env
```
Comment (EvolveArt) [inline infra/pragma-node/Dockerfile:29]: you can let it commented, env values are filled from parameter store on AWS

File: pragma-node/src/config.rs [hunk 5]
```diff
@@ -3,20 +3,23 @@ use serde::Deserialize;
 use tokio::sync::OnceCell;
 
 #[derive(Debug, Deserialize)]
-struct ServerConfig {
-    host: String,
-    port: u16,
+pub struct ServerConfig {
+    pub host: String,
```
Comment (EvolveArt) [inline pragma-node/src/config.rs:7]: let them private and create getters

File: pragma-node/src/infra/repositories/onchain_repository.rs [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+            AP.aggregated_price
+        FROM 
+            FilteredEntries FE,
+            AggregatedPrice AP
+        ORDER BY 
+            FE.timestamp DESC;
+    "#,
+        table_name = get_table_name_from_network(network),
+        backward_interval = BACKWARD_TIMESTAMP_INTERVAL,
+        aggregation_subquery = get_aggregation_query(aggregation_mode).unwrap()
```
Comment (EvolveArt) [inline pragma-node/src/infra/repositories/onchain_repository.rs:94]: remove unwrap

**Review by EvolveArt — COMMENTED** [review]
left few comments

File: pragma-node/src/infra/repositories/onchain_repository.rs (hunk 1 above)
Comment (akhercha) [reply pragma-node/src/infra/repositories/onchain_repository.rs]: yup true - it was to adhere to the `SpotEntry` struct but commented out this section for now, instead we take the last source timestamp as `last_updated_timestamp`

File: pragma-node/src/infra/repositories/onchain_repository.rs (hunk 6 above)
Comment (akhercha) [reply pragma-node/src/infra/repositories/onchain_repository.rs]: removed!

File: pragma-node/src/config.rs (hunk 5 above)
Comment (akhercha) [reply pragma-node/src/config.rs]: ah we no longer need those getters - leftover from when I added the onchain JsonRpcClient

removed the pub!

File: infra/pragma-node/Dockerfile (hunk 4 above)
Comment (akhercha) [reply infra/pragma-node/Dockerfile]: ah yesss, reverted

File: infra/pragma-node/config.yml (hunk 3 above)
Comment (akhercha) [reply infra/pragma-node/config.yml:9]: added!

File: infra/pragma-ingestor/config.yml (hunk 2 above)
Comment (akhercha) [reply infra/pragma-ingestor/config.yml]: changed!

**Review by EvolveArt — APPROVED** [review]

**Pull request merged**